app/assets/javascripts/host_edit.js

```diff
@@ -283,6 +283,18 @@ function update_form(element, options) {
   })
 }
 
+function serializeForm() {
+  return $($('form')[0].elements).filter(isExplicit).serialize()
+}
+
+function isExplicit(index, element) {
```

Comment:
>What is this doing exactly? Maybe a comment would be necessary here.

Basically, I am trying to filter out inputs that  don't have any value set on them explicitly.

>What is element, form elements? If so, why is it calling nextSibling on each ot them?

element is form input element, for elements that have "inherit" button (like environment and compute profile), there is a [flag](https://github.com/theforeman/foreman/blob/develop/app/helpers/layout_helper.rb#L174) set on the button itself that says that the value is set explicitly. I am trying to filter out the inputs that this flag is not set on them.

>Is there a way to avoid that chain of if checks?

Not that I am aware of. I have no knowledge about which input gets into the function, so I need to defend myself from any deviation in the hierarchy. The only idea that I have is using `try...catch` block, but it smells bad.